### PR TITLE
CRUD Pages for Pattern path

### PR DESCRIPTION
**Pull request by burakgzt**
Updates pattern path pages with new CRUD components + Adds dropdown functionality to add&edit component and cascaded list field functionality to list component

### CODE REVIEW

File: web/models/PatternPathLocationModel.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { ModelType, FormDataType } from './Models';
+
+export const PatternPathLocationModel: ModelType = {
+    tableName: 'pattern_path_location',
+    resolverName: 'PatternPathLocation',
+
+    endpoints: {
+        list: 'patternPathLocations',
+        export: '',
```
Comment (LouisAmon) [inline web/models/PatternPathLocationModel.tsx:9]: Not very elegant

Is it possible to not specify the `export` attribute at all instead ?

File: web/models/PatternPathLocationModel.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { ModelType, FormDataType } from './Models';
+
+export const PatternPathLocationModel: ModelType = {
+    tableName: 'pattern_path_location',
```
Comment (LouisAmon) [inline web/models/PatternPathLocationModel.tsx:4]: Is that the PostgreSQL table name ?

Why would anybody working on the frontend need to know that ?

File: web/pages/pattern-paths/add/index.tsx [hunk 3]
```diff
@@ -1,16 +1,114 @@
...
+    
+    /*useEffect(() => {
+        if (stockOwnerData.data) {
+            const newIdOpts: Array<FormOptionType> = [];
+            stockOwnerData.data.stockOwners?.results.forEach(({ id, name }) => {
+                newIdOpts.push({ text: name!, key: id! });
+            });
+            setSIdOptions(newIdOpts);
+        }
+    }, [stockOwnerData.data]);*/
```
Comment (LouisAmon) [inline web/pages/pattern-paths/add/index.tsx:67]: Can't leave that comment in production ;)

File: web/models/PatternPathLocationModel.tsx (hunk 2 above)
Comment (burakgzt) [reply web/models/PatternPathLocationModel.tsx]: It was required for displaying items with respect to permission and it was same before the CRUD components. However, we may not need to type it actually. I updated the code again to import it from GraphQL codegen and removed the hard-coded version

**Review by LouisAmon — APPROVED** [review]
Made some comments but I think we can overall validate it

**Pull request merged**